drivers/eeprom/eeprom_i2c.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2019 Nick Brassel (tzarc)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/*
+    The base I2C address of the EEPROM.
+    This needs to be shifted up by 1, to match i2c_master requirements.
+*/
+#ifndef EXTERNAL_EEPROM_I2C_BASE_ADDRESS
+#    define EXTERNAL_EEPROM_I2C_BASE_ADDRESS 0b10100000
+#endif
+
+/*
+    The calculated I2C address based on the input memory location.
+
+    For EEPROM chips that embed part of the memory location in the I2C address
+    such as AT24M02 you can use something similar to the following (ensuring the
+    result is shifted by left by 1):
+
+    #define EXTERNAL_EEPROM_I2C_ADDRESS(loc) \
+        (EXTERNAL_EEPROM_I2C_BASE_ADDRESS | ((((loc) >> 16) & 0x07) << 1))
+
```

Comment:
Would like to see the adafruit and sparkfun modules defaulted here, so that it's easy to set up. 

```suggestion

#if defined(EEPROM_I2C_SPARKFUN_2) || defined(EEPROM_I2C_RM24C512C)
#    define EXTERNAL_EEPROM_BYTE_COUNT 65536
#    define EXTERNAL_EEPROM_PAGE_SIZE 128
#    define EXTERNAL_EEPROM_ADDRESS_SIZE 2
#    define EXTERNAL_EEPROM_WRITE_TIME 5
#elif defined(EEPROM_I2C_MB85RC256V) || defined(EEPROM_I2C_ADAFRUIT_FRAM)
#    define EXTERNAL_EEPROM_BYTE_COUNT 32768
#    define EXTERNAL_EEPROM_WRITE_TIME 0
#endif
```

This way, you can add a simple